### DISCUSSION TITLE
How to pkg update without any prompt

### DISCUSSION THREAD

**Discussion by khansaad1275** [termux/termux-app#2652] [Q&A]
Whenever I am trying to create a script I update the termux packages, and since it is a script i want everything to happen automatically without any prompt, so I use `pkg update -y`, this way it doesn't ask me for (Y/n) But it always asks me for openssl.cnf ( as you can see in the below image)

![image](https://user-images.githubusercontent.com/32749921/157847658-57e796b2-ec86-46e8-afa7-2e077c4d1ab8.png)

**Is There any way I can bypass this too? And always go with the N ( default ) option.** 

- Thank you.

**Comment by agnostic-apollo** [comment]
Use `apt-get install -y -o Dpkg::Options::="--force-confdef" -o Dpkg::Options::="--force-confold" <package_name>` command to install/update the packages. The `-y` is to assume `yes` as answer to all prompts and run non-interactively. The dpkg options are related to conf files. During updates, any conf file that has been previously modified by the user will not be replaced with the new version provided by updated package and no prompts will be shown to allow for command to run non-interactively. Check `https://manpages.debian.org/jessie/dpkg/dpkg.1.en.html` for more info.

If your script runs in background/non-interactively, then you shouldn't be using `pkg` since its a wrapper for `apt` and should use `apt-get` instead. You can run `apt-get update` if not using `pkg` to update apt cache before `apt-get install`, `pkg` does that automatically.

https://github.com/termux/termux-packages/blob/master/packages/termux-tools/pkg
